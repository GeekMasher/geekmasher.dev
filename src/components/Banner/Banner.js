// @flow
import React from 'react';
import { get } from 'lodash';
import styles from './Banner.module.scss';
import Meta from '../Meta';

type Props = {
  created: string,
  updated: string
};

const Banner = ({ banner, created, updated }: Props) => {

  const link = get(banner, 'link');
  const caption = get(banner, 'caption');

  return (
    <div className={styles['banner']}>
      <img src={get(banner, 'path')} alt={get(banner, 'caption')} />
      { link !== undefined && caption !== undefined &&
      <p className={styles['banner__caption']}>
        <i>
          <a href={link} target="_blank" rel="noopener noreferrer">
          &copy; {caption}
          </a>
        </i>
      </p>
      }
      <Meta created={created} updated={updated} />
    </div>
  );
};

export default Banner;
