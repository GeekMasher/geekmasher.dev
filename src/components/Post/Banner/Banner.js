// @flow
import React from 'react';
import moment from 'moment';
import styles from './Banner.module.scss';
import Meta from '../Meta';

type Props = {
  path: string,
  caption: string,
  created: string,
  updated: string
};

const Banner = ({ path, caption, created, updated }: Props) => (
  <div className={styles['banner']}>
    <img src={path} alt={caption} />
    <p className={styles['banner__caption']}>
      <i>{caption}</i>
    </p>
    <Meta publishDate={created} lastUpdated={updated} />
  </div>
);

export default Banner;
