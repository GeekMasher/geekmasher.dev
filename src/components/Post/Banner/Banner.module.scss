@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.banner {
    max-width: $layout-post-single-width;
    margin-left: auto;
    margin-right: auto;
    padding-top: 1rem;

    & img {
        max-width: 100%;
        margin: 0 auto;
    }
    &__caption {
        max-width: 100%;
        text-align: right;
        margin-top: 0.1rem;
    }
}