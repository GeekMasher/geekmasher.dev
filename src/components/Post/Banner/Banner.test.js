// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import Banner from './Banner';

describe('Banner', () => {
  it('renders correctly', () => {
    const props = {
        created: '2019-05-01',
        path: '/media/sample.jpg',
        caption: 'test'
    };

    const tree = renderer.create(<Banner {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
