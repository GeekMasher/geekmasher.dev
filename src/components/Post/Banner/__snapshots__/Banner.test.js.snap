// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders correctly 1`] = `
<div
  className="banner"
>
  <img
    alt="test"
    src="/media/sample.jpg"
  />
  <p
    className="banner__caption"
  >
    <i>
      test
    </i>
  </p>
  <div
    className="meta"
  >
    <p
      className="meta__date"
    >
      <b>
        1 May 2019
      </b>
    </p>
  </div>
</div>
`;
